feat: add release history

GlennBullingham opened the pull request:
- adds a Release Notes section to the TOC
 - adds a version history page

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
feat: add release history

 - adds a Release Notes section to the TOC
 - adds a version history page

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

anupriya0703 commented on SUMMARY.md:
```diff
@@ -1,6 +1,10 @@
 # Table of contents
 
-* [Welcome to Mayastor!](README.md)
+* [Welcome to Mayastor!](README.md)]
```
There is an extra ']'.

GlennBullingham replied:
Good spot, thank you.  I've corrected it.

GlennBullingham committed:
fix(release_notes): typo and formating

 - fixes a typo in the TOC file
 - resolves formatting issues with history.md

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham commented:
In addition to the typo in the TOC file, there were also unsaved changes on my local branch relating to the formatting of the history.md file which I have now pushed, so you may wish to take another look there too.  Thanks.

avishnu commented on releases/history.md:
```diff
@@ -0,0 +1,13 @@
+  * released: 11-Oct-2022
+* ### Patch Releases:
+  * [v1.0.1](https://github.com/openebs/mayastor/releases/tag/v1.0.1)
+  * [v1.0.2](https://github.com/openebs/mayastor/releases/tag/v1.0.2)
```
Should v1.0.2 precede v1.0.1 ? Latest at the top, followed by others with oldest at the bottom?

GlennBullingham replied:
I'll be honest, I was copying from kubernetes.io homework.  They list in a similar fashion, using ascending order.  So I'm "following suit" with something that users will have seen elsewhere.

GlennBullingham replied:
However, with time to consider it, I think it is logical to go in reverse chronological order here.  I'll update.

GlennBullingham committed:
refactor: list latest patch versions first

Use reverse chronological order in patch versions
list of the release history page

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

avishnu reviewed (approved):
lgtm
